### PR TITLE
8342860: Fix more NULL usage backsliding

### PR DESCRIPTION
**Pull request by theoweidmannoracle**
- Changed several "NULL" in comments to "null"
- Changed several `NULL` in code to `nullptr`

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8342860](https://bugs.openjdk.org/browse/JDK-8342860): Fix more NULL usage backsliding (**Enhancement** - P4)

### Reviewers
 * [Johan Sjölen](https://openjdk.org/census#jsjolen) (@jdksjolen - **Reviewer**) Review applies to [9754145b](https://git.openjdk.org/jdk/pull/21826/files/9754145be5be68284338fb5995c778685b8cad7f)
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Julian Waters](https://openjdk.org/census#jwaters) (@TheShermanTanker - Committer)
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21826/head:pull/21826` \
`$ git checkout pull/21826`

Update a local copy of the PR: \
`$ git checkout pull/21826` \
`$ git pull https://git.openjdk.org/jdk.git pull/21826/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21826`

View PR using the GUI difftool: \
`$ git pr show -t 21826`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21826.diff">https://git.openjdk.org/jdk/pull/21826.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21826#issuecomment-2457162493)
</details>

### CODE REVIEW

File: src/hotspot/os/linux/cgroupSubsystem_linux.cpp
Comment (SoniaZaldana) [inline src/hotspot/os/linux/cgroupSubsystem_linux.cpp:1]: Missing copyright update.

**Review by SoniaZaldana — CHANGES_REQUESTED** [review]
I am not a Reviewer but I left a small comment. 

Cheers!

File: src/hotspot/os/linux/cgroupSubsystem_linux.cpp
Comment (theoweidmannoracle) [reply src/hotspot/os/linux/cgroupSubsystem_linux.cpp:1]: Thanks for spotting this!

**Comment by SoniaZaldana** [comment]
Hi @theoweidmannoracle, I think the GHA tests are not running because you haven't enabled GHA on your personal fork. See https://wiki.openjdk.org/display/SKARA/Testing for a bit more info.

**Comment by theoweidmannoracle** [comment]
Hi @SoniaZaldana, thanks for pointing that out. I enabled the tests now, but it seems there's no way to run them except for pushing new changes. I did run Oracle's internal testing, though.

**Comment by SoniaZaldana** [comment]
> Hi @SoniaZaldana, thanks for pointing that out. I enabled the tests now, but it seems there's no way to run them except for pushing new changes. I did run Oracle's internal testing, though.

You can try syncing your fork and that should trigger GHA.

**Comment by theoweidmannoracle** [comment]
@SoniaZaldana Thanks for the tip!

**Review by kimbarrett — CHANGES_REQUESTED** [review]
Can you use the (updated) regex in the JBS issue description to verify the only
remaining "NULL"s in src/hotspot are jvmti.{xml,xsl} and
globalDefinitons_{gcc,visCPP}.hpp?

There are also some new NULLs in test/hotspot.
./jtreg/serviceability/jvmti/GetMethodDeclaringClass/libTestUnloadedClass.cpp
./jtreg/serviceability/jvmti/vthread/VThreadEventTest/libVThreadEventTest.cpp
There are a couple more (after filtering out java and C source files) that I think
shouldn't be changed.

**Review by TheShermanTanker — APPROVED** [review]

**Review by kimbarrett — CHANGES_REQUESTED** [review]
Can you use the (updated) regex in the JBS issue description to verify the
only remaining "NULL"s in src/hotspot are the jvmti.{xml,xls} files and the
globalDefinitions_{gcc,visCPP}.hpp files?

There are also some NULLs recently introduced in test/hotspot:
./jtreg/serviceability/jvmti/GetMethodDeclaringClass/libTestUnloadedClass.cpp
./jtreg/serviceability/jvmti/vthread/VThreadEventTest/libVThreadEventTest.cpp

(Found by applying the same regex to test/hotspot, and then removing .java and
.c files.)

There are a few other files in test/hotspot containing NULLs:
./jtreg/vmTestbase/nsk/share/jni/README
./jtreg/vmTestbase/nsk/share/jvmti/README
These are documentation files with examples written in C, so should not be changed.

./jtreg/vmTestbase/nsk/share/native/nsk_tools.hpp
In a comment describing a string to be used for printing.  Uses would need to
be examined to ensure it's okay to change the string used for a null value.
I think I planned to do this as a followup to JDK-8324799, and then forgot.
I'd be okay with doing something about this being separate from the current
PR.  While the necessary textual changes are probably small, there's a lot of
uses to examine to be sure a change is okay.

**Comment by kimbarrett** [comment]
> Hi @SoniaZaldana, thanks for pointing that out. I enabled the tests now, but it seems there's no way to run them except for pushing new changes. I did run Oracle's internal testing, though.

To manually trigger GHA tests:
1. Go to your personal fork, and click on the "Actions" menu item.
2. Select the "OpenJDK GHA Sanity Checks" Action.
3. Click on the "Run workflow" pulldown.
4. Select the branch you want to test in the "Use workflow from" pulldown.
5. Click on "Run workflow".

**Comment by theoweidmannoracle** [comment]
> Can you use the (updated) regex in the JBS issue description to verify the only remaining "NULL"s in src/hotspot are the jvmti.{xml,xls} files and the globalDefinitions_{gcc,visCPP}.hpp files?
> 
> There are also some NULLs recently introduced in test/hotspot: ./jtreg/serviceability/jvmti/GetMethodDeclaringClass/libTestUnloadedClass.cpp ./jtreg/serviceability/jvmti/vthread/VThreadEventTest/libVThreadEventTest.cpp
> 
> (Found by applying the same regex to test/hotspot, and then removing .java and .c files.)
> 
> There are a few other files in test/hotspot containing NULLs: ./jtreg/vmTestbase/nsk/share/jni/README ./jtreg/vmTestbase/nsk/share/jvmti/README These are documentation files with examples written in C, so should not be changed.
> 
> ./jtreg/vmTestbase/nsk/share/native/nsk_tools.hpp In a comment describing a string to be used for printing. Uses would need to be examined to ensure it's okay to change the string used for a null value. I think I planned to do this as a followup to JDK-8324799, and then forgot. I'd be okay with doing something about this being separate from the current PR. While the necessary textual changes are probably small, there's a lot of uses to examine to be sure a change is okay.

@kimbarrett I fixed the backslides in the *.cpp files you mentioned. The egrep outputs are now:

```
% find test/hotspot -type f ! -name "*.c" ! -name "*.java" -exec egrep -H "[^[:alnum:]_]NULL([^[:alnum:]_]|$)" {} \;
test/hotspot/jtreg/vmTestbase/nsk/share/native/nsk_tools.hpp: * Returns str or "<NULL>" if str is null; useful for printing strings.
test/hotspot/jtreg/vmTestbase/nsk/share/jvmti/README:    if (!NSK_JVMTI_VERIFY(jvmti->GetVersion(&version) != NULL)) {
test/hotspot/jtreg/vmTestbase/nsk/share/jni/README:            jni->FindClass(class_name) != NULL)) {
test/hotspot/jtreg/vmTestbase/nsk/share/jni/README:            jni->FindClass(class_name)) != NULL)) {
```

and 

```
egrep -R "[^[:alnum:]_]NULL([^[:alnum:]_]|$)" src/hotspot                                                      
src/hotspot/share/prims/jvmti.xml:    or return value. A "null pointer" is C <code>NULL</code> or C++ <code>nullptr</code>.
src/hotspot/share/prims/jvmti.xml:                       &amp;methodName, NULL, NULL);
src/hotspot/share/prims/jvmti.xml:  GetThreadCpuTime(env, NULL, nanos_ptr)
src/hotspot/share/prims/jvmti.xsl:        <xsl:value-of select="@const"/>, NULL)</code>
src/hotspot/share/utilities/globalDefinitions_visCPP.hpp:// When __cplusplus is defined, NULL is defined as 0 (32-bit constant) in
src/hotspot/share/utilities/globalDefinitions_visCPP.hpp:// On 64-bit architectures, defining NULL as a 32-bit constant can cause
src/hotspot/share/utilities/globalDefinitions_visCPP.hpp:// varargs, we pass the argument 0 as an int.  So, if NULL was passed to a
src/hotspot/share/utilities/globalDefinitions_visCPP.hpp:// only 32-bits of the "NULL" pointer may be initialized to zero.  The
src/hotspot/share/utilities/globalDefinitions_visCPP.hpp:// Solution: For 64-bit architectures, redefine NULL as 64-bit constant 0.
src/hotspot/share/utilities/globalDefinitions_visCPP.hpp:#undef NULL
src/hotspot/share/utilities/globalDefinitions_visCPP.hpp:#define NULL 0LL
src/hotspot/share/utilities/globalDefinitions_visCPP.hpp:#ifndef NULL
src/hotspot/share/utilities/globalDefinitions_visCPP.hpp:#define NULL 0
src/hotspot/share/utilities/globalDefinitions_visCPP.hpp:// NULL vs NULL_WORD:
src/hotspot/share/utilities/globalDefinitions_visCPP.hpp:// On Linux NULL is defined as a special type '__null'. Assigning __null to
src/hotspot/share/utilities/globalDefinitions_visCPP.hpp:#define NULL_WORD NULL
src/hotspot/share/utilities/globalDefinitions_gcc.hpp:// NULL vs NULL_WORD:
src/hotspot/share/utilities/globalDefinitions_gcc.hpp:// On Linux NULL is defined as a special type '__null'. Assigning __null to
src/hotspot/share/utilities/globalDefinitions_gcc.hpp:  #define NULL_WORD  NULL
```

**Review by kimbarrett — APPROVED** [review]
Looks good.

The grep results are exactly as expected.  Thanks for checking that.  Now if we can
just get the build to start checking for us, we can stop needing these periodic cleanups.
I forget whether a JBS issue has been filed for that.  If not, I will do so.

**Comment by kimbarrett** [comment]
> The grep results are exactly as expected. Thanks for checking that. Now if we can just get the build to start checking for us, we can stop needing these periodic cleanups. I forget whether a JBS issue has been filed for that. If not, I will do so.

II've filed these followup issues:
https://bugs.openjdk.org/browse/JDK-8343802 Prevent NULL usage backsliding
https://bugs.openjdk.org/browse/JDK-8343800 Cleanup definition of NULL_WORD
https://bugs.openjdk.org/browse/JDK-8343801 Change string printed by nsk_null_string for null strings

**Comment by theoweidmannoracle** [comment]
@jdksjolen @TheShermanTanker It would be great if you could also take another look at the latest changes I made. Thanks!

**Review by TheShermanTanker — APPROVED** [review]

**Review by tschatzl — APPROVED** [review]

**Pull request closed without merging**